### PR TITLE
.github/workflows/doc-build.yml: Fix generation of CHANGES.html

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

This new feature from #35652 was supposed to show the changed HTML files, 
but a last minute change there destroyed it.

Here we fix it.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -80,7 +80,10 @@ jobs:
         # incremental docbuild may introduce broken links (inter-file references) though build succeeds
         run: |
           set -ex
-          make doc-clean doc-uninstall sagelib-clean && git clean -fx src/sage && ./config.status && make doc-html
+          mv /sage/local/share/doc/sage/html/en/.git /sage/.git-doc
+          make doc-clean doc-uninstall sagelib-clean && git clean -fx src/sage
+          mkdir -p /sage/local/share/doc/sage/html/en/ && mv /sage/.git-doc /sage/local/share/doc/sage/html/en/
+          ./config.status && make doc-html
```
Comment (kwankyu) [inline .github/workflows/doc-build.yml:86]: Move `.git-doc` to `.git-doc`? No `.git`?

Where is the code to generate `CHANGES.html`? No idea how this works.

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:86]: Thanks, fixed now.

`CHANGES.html` is generated in the "Copy docs" step

Comment (kwankyu) [reply .github/workflows/doc-build.yml:86]: Are you aware that L53 is very similar to L100 in the "Copy docs" step. This seems to be a mistake to me...

Comment (kwankyu) [reply .github/workflows/doc-build.yml:86]: Yes, you are.

Comment (kwankyu) [reply .github/workflows/doc-build.yml:86]: What is the net effect of L83 and L85. It is moving `.git` directory to and from `.gitdoc`. Why?

Comment (kwankyu) [reply .github/workflows/doc-build.yml:86]: In particular, why `mkdir -p /sage/local/share/doc/sage/html/en/`. The directory already exists. Why create it again?

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:86]: I'm moving the `.git` directory away and back to protect it from `make doc-uninstall`

Comment (kwankyu) [reply .github/workflows/doc-build.yml:86]: OK.

**Comment by kwankyu** [comment]
Most of the files in `changes` in the documentation preview of this PR still seem spurious. There should be none?

**Comment by mkoeppe** [comment]
I think they are all caused by functions whose default arguments display a hex address

**Comment by kwankyu** [comment]
I don't know if wiping out sage version from the sidebar is a good idea... 

If I would change it at all and if possible, then I would augment the version number with the PR number, like `Sage 10.1.beta4#12345` from `Sage 10.1.beta4`

**Comment by kwankyu** [comment]
By second thought, I would not change the version number at all because checking if the version number appear correct in the documentation is itself a part of the purpose of the documentation preview.

**Comment by kwankyu** [comment]
> I think they are all caused by functions whose default arguments display a hex address

Yeah. I see.

**Comment by kwankyu** [comment]
And how about providing diff separately for each changed html? 

If there is a free API service that show git diff nicely, then we may use that...

**Comment by mkoeppe** [comment]
> I would not change the version number at all because checking if the version number appear correct in the documentation is itself a part of the purpose of the documentation preview.

Without making the version number equal between both versions, every single file would show up in the diff

**Comment by kwankyu** [comment]
Both should show the same latest beta version. No? But  I guess there could be a mismatch, for example, for some short time before the latest beta appears in the docker image, or when the branch of a PR is not rebased onto the latest beta. Is your making-version-number-equal a provision for these situations?

**Comment by mkoeppe** [comment]
Also when the branch of the PR has not merged the latest beta.

**Comment by kwankyu** [comment]
I agree that they are common. But if the author recognizes the version mismatch, then he/she can just upload the branch merged with the latest beta, which I think is, anyway, a good thing to do. Also this is just optional.

On the other hand, we frequently create links to the doc built for a PR, then it is good that the doc is accompanied with the right beta version.

I am not sure about this matter though. I would not insist.

**Comment by mkoeppe** [comment]
I'll change it so that the old copy gets updated with the version number of the PR branch.

**Comment by kwankyu** [comment]
Thanks. It seems a good compromise.

**Comment by kwankyu** [comment]
This is of course a wishlist item.

> And how about providing diff separately for each changed html, instead of one giant diff `HTML.diff`?
> 
> If there is a free API service that show git diff nicely, then we may use that...

There is https://diffy.org/

**Review by kwankyu — APPROVED** [review]
It works now.

**Comment by mkoeppe** [comment]
Thanks for the review!

**Comment by tobiasdiez** [comment]
> This is of course a wishlist item.
> 
> > And how about providing diff separately for each changed html, instead of one giant diff `HTML.diff`?
> > If there is a free API service that show git diff nicely, then we may use that...
> 
> There is https://diffy.org/. But it seems not work with a url.

It wouldn't be too hard to use https://diff2html.xyz/

**Pull request merged**